test/prob/discrete_range/discrete_range_test.hpp

```diff
@@ -0,0 +1,62 @@
+// Arguments: Ints, Ints, Ints
+#include <stan/math/prim.hpp>
+
+using std::vector;
+
+class AgradDistributionsHypergeometric : public AgradDistributionTest {
```

Comment:
The name should match what's being tested.  

We should also follow uniform_lpdf and throw if the parameters are not properly ordered.

I believe we should also be throwing a domain error when there's a domain error, that is when the variant isn't within the bounds.  That's not how uniform_lpdf or pareto_lpdf currently work, so I'm going to open another issue to fix those and this one.

I also believe we should be throwing when the variant is not in domain.  That's what we do for other distributions.  The `uniform_lpdf`, but it should be.  

Now that I see these tests, I realize that we should be throwing when there's a domain error, not just returnign -infinity.  For instance, consider `uniform_lpdf`, which throws if the parameters aren't properly ordered.  I'm going to open yet another issue to fix all this in the other distributions.